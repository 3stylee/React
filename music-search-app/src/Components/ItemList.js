import React from "react"
import Pagination from "./Pagination"
import { Typography, Box } from "@mui/material"
import ItemCard from "./ItemCard"

const ItemList = props => {
    if (props.results.length > 0){
        return (
            <>
                {props.results.map((item, index) => (<ItemCard key={index} data={item} color={props.color}></ItemCard>))}
                <Box mt={4}>
                    <Pagination page={props.page} onPageChanged={props.onPageChanged} />
                </Box>
            </>
        )
    } else {
        return (<Typography variant="h5" component="h5">Sorry, no results found</Typography>)
    }
}

export default ItemList;