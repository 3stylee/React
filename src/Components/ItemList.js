import React from 'react';
import { Typography, Grid, CircularProgress, Button } from '@mui/material';
import ItemCard from './ItemCard';
import { StyledGrid } from './StyledComponents';
import { useInView } from 'react-intersection-observer';
import { useEffect } from 'react';

const ItemList = props => {
	const [ref, inView] = useInView();
	useEffect(() => {
		if (inView) {
			props.onLoadMore();
		}
	}, [inView]);

	if (props.initial) {
		return;
	}
	if (props.results.length > 0 || props.loading) {
		return (
			<>
				<StyledGrid container spacing={2}>
					{props.results.map((item, index) => (
						<Grid item xs={12} sm={6} md={4} lg={3} key={index}>
							<ItemCard data={item} />
						</Grid>
					))}
				</StyledGrid>
				{props.loading ? (
					<CircularProgress color="inherit" />
				) : (
					<Button
						ref={ref}
						onClick={props.onLoadMore}
						color="secondary">
						Load More...
					</Button>
				)}
			</>
		);
	} else {
		return (
			<Typography variant="h5" component="h5">
				Sorry, no results found
			</Typography>
		);
	}
};

export default ItemList;
