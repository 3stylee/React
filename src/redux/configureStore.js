import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './reducers';
import thunk from 'redux-thunk';

const setUpStore = initalState => {
	return configureStore({
		reducer: rootReducer,
		preloadedState: initalState,
		middleware: [thunk],
	});
};

export default setUpStore;
