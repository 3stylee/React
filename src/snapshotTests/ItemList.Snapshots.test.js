import React from 'react';
import ItemList from '../Components/ItemList';
import renderer from 'react-test-renderer';
import { results as mockResults } from '../snapshotTests/mockData';
import { Provider } from 'react-redux';
import configureStore from '../redux/configureStore';

it('displays a circular progress indicator when loading is set to true', () => {
	const tree = renderer.create(
		<ItemList
			results={mockResults}
			loading={true}
			constructURL={jest.fn()}
			initial={false}
		/>
	);

	expect(tree).toMatchSnapshot();
});

it('displays nothing if initial is set to true', () => {
	const tree = renderer.create(
		<ItemList
			results={mockResults}
			loading={false}
			constructURL={jest.fn()}
			initial={true}
		/>
	);

	expect(tree).toMatchSnapshot();
});

it("displays 'Sorry, no results found' if the results length is not greater than 0", () => {
	const tree = renderer.create(
		<ItemList
			results={{}}
			loading={false}
			constructURL={jest.fn()}
			initial={false}
		/>
	);

	expect(tree).toMatchSnapshot();
});

it('displays a 4x3 grid of cards if the results are populated', () => {
	const store = configureStore();

	const tree = renderer.create(
		<Provider store={store}>
			<ItemList
				results={mockResults}
				loading={false}
				constructURL={jest.fn()}
				initial={false}
			/>
		</Provider>
	);

	expect(tree).toMatchSnapshot();
});
