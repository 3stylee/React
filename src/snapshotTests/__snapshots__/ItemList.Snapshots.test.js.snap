// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays 'Sorry, no results found' if the results length is not greater than 0 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
>
  Sorry, no results found
</h5>
`;

exports[`displays a 4x3 grid of cards if the results are populated 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Adele
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            artist
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              30
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Adele
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele Bloemendaal
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Skyfall - Single
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            track
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Skyfall - Single
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Emeli Sandé
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Emeli Sandé
            , Media Type:
             
            artist
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              30
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            track
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Our Version of Events
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Emeli Sandé
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              21
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              25
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            album
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              30
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            Adele
            , Media Type:
             
            track
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1plj4ik-MuiPaper-root-MuiCard-root"
        color="#e3f2fd"
      >
        <div
          className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              Mad Skillet
            </span>
          </div>
        </div>
        <div
          className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            Artist: 
            John Medeski's Mad Skillet
            , Media Type:
             
            track
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiBox-root css-178yklu"
  >
    <div
      aria-label="outlined button group"
      className="MuiButtonGroup-root MuiButtonGroup-outlined css-iajp3t-MuiButtonGroup-root"
      role="group"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-1l9sosn-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        1
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-1l9sosn-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        2
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-1l9sosn-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        3
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedSecondary css-1l9sosn-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        ...
      </button>
    </div>
  </div>,
]
`;

exports[`displays a circular progress indicator when loading is set to true 1`] = `
<span
  className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
  role="progressbar"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`displays nothing if initial is set to true 1`] = `null`;
